docs/reusing-typechecking-results.md

```diff
@@ -0,0 +1,171 @@
+---
+title: Reusing typechecking results
+category: Compiler Internals
+categoryindex: 200
+index: 42
+---
+
+# Reusing typechecking results between compiler runs
+
+Caching and reusing compilation results between compiler runs will be an optimization technique aimed at improving the performance and efficiency of the F# development experience.
+
+## Motivation
```

Comment:
I wanted this as the ultimate endgame - it will feel like the build is instant (because it would be running while you type, and have plenty of time before you move to triggering a test run).

But the options are not the same neither are the runtimes (tooling right now still on desktop), so we need to remove the big obstacles first.

Reply:
Still might be a good time to start thinking about it. Maybe start by including some kind of versioning which will let us know whether given cached results can be used or not. And it can initially only be available in some subset of cases.

Reply:
Will resolve it then. We are keeping the endgoal in our minds!